### DISCUSSION TITLE
[extendTheme] Help with globally extending colors in the default theme

### DISCUSSION THREAD

**Discussion by YashKarthik** [chakra-ui/chakra-ui#5260] [Q&A] [answered]
Hi. I've been trying to figure out a way to implement dark mode (globally) without having to call `useColorModeValue('light', 'dark')` for each component.

In my `theme.js` I have the following styling. [From the chakra docs](https://chakra-ui.com/docs/theming/customize-theme#customizing-global-styles)

```Javascript
const overrides = extendTheme({
  styles: {
    global: (props) => ({
      body: {
        fontFamily: 'body',
        color: mode('gray.800', 'whiteAlpha.900')(props),
        bg: mode('white', 'gray.800')(props),
        lineHeight: 'base',
      },
    }),
  },
})
```
This allows me to access the colors in `Header.js` like this:
```JSX
<Text textColor='styles.body.color'>{props.content}</Text>
```

How can I go about adding more color options to this? I tried this for the `link`, `a` & `p` tags (it didn't work):

```JSX
const overrides = extendTheme({
  styles: {
    global: (props) => ({
      body: {
        ...
        link: mode('blue.600', 'blue.300')(props)
      },
    }),
  },
})
```

I also tried putting the styles for `link` etc outside the `body` object. None of them worked.

Thanks :)

**Comment by LucasZapico** [comment]
Hey 👋 , 

I'm not sure if I should open up an issue or separate discussion. But my issue is similar if not identical. I theme my app/sites in the theme object then use branded colors for typography and custom components. 

__Custom theming example__ 

*color.js*. 
```javascript 
export const Colors = {
    black: '#000000',
    white: '#FFFFFF',
    brand: {
      zero: '#5C7AEA',
      one: '#3D56B2',
      two: '#CEE5D0', 
      three: '#406343', 
      four: '#32502E', 
      five: '#79B4B7', 
    },
  },
```
**typography.js**
```javascript
export const Texts = {
  //....
  variants: {
    mono: {
      fontFamily: 'Roboto Mono',
    },
    sec: {
      color: 'brand.one,
    },
    tri: {
      color: 'brand.two',
    },
  },
};
```

## Desired Functionality

What I want is to be able to use color mode props in my other components allowing mode theming at the theme level. I believe this aligns with the value and philosophy of Chakra UI. 

__Example__ 

```javascript 
const AppTheme = extendTheme({
  config,
  components: (props) => ({
    
    Text: {
      variants: {
      mono: {
        fontFamily: 'Roboto Mono',
      },
      sec: {
        color: props.colorMode === 'dark' ? '#fbfbfb' : '#333333',
      },
      tri: {
        color: props.colorMode === 'dark' ? 'brand.one' : 'brand.three',
      },
    },
  }
  })
})
```

Ideally, the theme props would be available to all components thus allowing for mode styling at the config obj level rather than in the components. Maybe I'm wrong but trying to delegate as much styling decision into the theme object reduced the probability of on-the-fly styling variation throughout a project. 

I have tried a few variations to try and get the theme level toggling but nothing has worked so far. Such as two themes toggled when the theme is passed in the provider. But we can't use context properties in another context. It has been a little and I might take another stab at working around it but whatever I solve will probably be a hack. 

Please let me know if this is unclear or if this should be a separate issue/discussion. 

Cheers,

**Comment by primos63** [comment, marked as answer]
@YashKarthik You do realize there is already a light and dark mode setting available to you via the theme.config settings detailed in the [docs](https://chakra-ui.com/docs/features/color-mode#updating-the-theme-config)? Not sure why you're looking to set everything in a global style, but you can also change the color of components by extending the theme with your modifications. If you've read through the [Theming section](https://chakra-ui.com/docs/theming/theme) of the documentation and something isn't clear please let me know.

@LucasZapico It's always best to provide an example via CodeSandbox of what you have that is not working. You can do it either with [JavaScript](https://codesandbox.io/s/chakra-ui-javascript-starter-template-flkwz) or [TypeScript](https://codesandbox.io/s/chakra-ui-typescript-starter-template-1my52).

**Reply by LucasZapico** [reply]
Hey @primos63 I am aware there is a default in config and I have explored pretty deeply how theming can be implemented. 


```javascript 
const config = {
  initialColorMode: 'light',
  useSystemColorMode:  true,
};
```

Unless I am misunderstanding the philosophy behind top-level styles configs. Isn't the value to ensure style consistency in the project and guard rails again minor style variation at the component level. In my opinion, if I define styles for a default mode in the theme it follows to define dark or any other mode at the theme level as well.  It is possible that this does not align with the principles of Chakra, which would be a misunderstanding on my part. 

I'll sit down and make a few Codesandbox's with theming variations that I have attempted. 

Thank you for the response. I love this project but this has been needling me for a while now. 

Cheers,

**Reply by YashKarthik** [reply]
Hey @primos63 I guess I wasn't clear in my original post. I know that Chakra ships with a default dark mode. I wanted to change the colors used in the dark mode.

I found this from [this discussion](https://github.com/chakra-ui/chakra-ui/discussions/5093#discussioncomment-1652354). Everything works now *except* the background colors. How do I change the default background color from `gray.800` to something else.

Thanks!

**Reply by YashKarthik** [reply]
Another update. I figured out how to change the background color as well.

The answer in [the same thread](https://github.com/chakra-ui/chakra-ui/discussions/5093#discussioncomment-1652354) works perfectly.

Cheers.

**Reply by primos63** [reply]
@YashKarthik Sorry it was not clear to me.  Yes, the solution in the thread would be what I would have suggested. Glad you were able to find it! Just don't forget some components have variants and you need to account for the colors there as well. The solution doesn't show modifying the color for `Button` which uses a default `colorScheme` and has colors for its variants and pseudo classes like _hover, _disabled. You can see this in the theme for [Button](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/button.ts).

**Reply by YashKarthik** [reply]
Got it.
Thanks a lot!
